src/main/java/uk/ac/ebi/subs/fileupload/services/globus/GlobusService.java

```diff
@@ -0,0 +1,359 @@
+package uk.ac.ebi.subs.fileupload.services.globus;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.data.mongodb.core.FindAndModifyOptions;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.stereotype.Service;
+import uk.ac.ebi.subs.data.fileupload.FileStatus;
+import uk.ac.ebi.subs.fileupload.services.EventHandlerService;
+import uk.ac.ebi.subs.fileupload.util.FileSource;
+import uk.ac.ebi.subs.fileupload.util.Utils;
+import uk.ac.ebi.subs.repository.model.Submission;
+import uk.ac.ebi.subs.repository.model.fileupload.File;
+import uk.ac.ebi.subs.repository.model.fileupload.GlobusShare;
+import uk.ac.ebi.subs.repository.repos.SubmissionRepository;
+import uk.ac.ebi.subs.repository.repos.fileupload.FileRepository;
+import uk.ac.ebi.subs.repository.repos.fileupload.GlobusShareRepository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class GlobusService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GlobusService.class);
+
+    @Autowired
+    private GlobusShareRepository globusShareRepository;
+
+    @Autowired
+    private FileRepository fileRepository;
+
+    @Autowired
+    private SubmissionRepository submissionRepository;
+
+    @Autowired
+    private EventHandlerService eventHandlerService;
+
+    @Autowired
+    private MongoOperations mongoOperations;
+
+    @Autowired
+    private GlobusApiClient globusApiClient;
+
+    @Value("${file-upload.sourceBasePath}")
+    private String sourceBasePath;
+    @Value("${file-upload.targetBasePath}")
+    private String targetBasePath;
+
+    @Value("${file-upload.globus.shareUrlStringFormat}")
+    private String shareUrlFormat;
+
+    @Value("${file-upload.globus.baseUploadDirectory}")
+    private String baseUploadDir;
+    @Value("${file-upload.globus.hostEndpoint.baseDirectory}")
+    private String hostEndpointBaseDir;
+
+    @Value("${file-upload.filePrefixForLocalProcessing}")
+    private String filePrefixForLocalProcessing;
+
+    public String getShareLink(String owner, String submissionId) {
+        return getOrCreateGlobusShare(owner, submissionId).getShareLink();
+    }
+
+    public void processUploadedFiles(String owner, String submissionId, List<String> files) {

```

Comment:
this method kicks off lots of file manipulation activities that will be cumbersome to test. this step is similar to tus's post finish  event handling where files are moved around directories and checksum and content validation processes are started. so i think we dont need tests for this.

Reply:
do we have an integration test for globus file upload?

Reply:
as far as i know, there is no integration testing being done for tus as well except for our api tester which is basically just automated testing. we can look into updating the api tester for testing globus file uploading but it will be a major undertaking and due to the way globus file uploading works will be complicated to setup as well (https://www.globus.org/globus-connect-personal)